pyqtorch/noise.py

```diff
@@ -0,0 +1,368 @@
+        px = torch.tensor(probabilities[0], dtype=DEFAULT_MATRIX_DTYPE)
+        py = torch.tensor(probabilities[1], dtype=DEFAULT_MATRIX_DTYPE)
+        pz = torch.tensor(probabilities[2], dtype=DEFAULT_MATRIX_DTYPE)
+        K0: Tensor = torch.sqrt(1.0 - (px + py + pz)) * IMAT
+        K1: Tensor = torch.sqrt(px) * XMAT
+        K2: Tensor = torch.sqrt(py) * YMAT
+        K3: Tensor = torch.sqrt(pz) * ZMAT
+        Kraus_PauliChannel: list[Tensor] = [K0, K1, K2, K3]
+        super().__init__(Kraus_PauliChannel, target, probabilities)
+
+
+class AmplitudeDamping(Noise):
+    """
+    Initialize the AmplitudeDamping gate.
+
+    The amplitude damping channel is defined as:
+
+    .. math::
+        \\rho \\Rightarrow K_0 \\rho K_0^{\\dagger} + K_1 \\rho K_1^{\\dagger}
+
+    with:
+
+    .. code-block:: python
+
+        K0 = [[1, 0], [0, sqrt(1 - rate)]]
+        K1 = [[0, sqrt(rate)], [0, 0]]
+
+    Args:
+        target (int): The index of the qubit being affected by the noise.
+        rate (float): The damping rate, indicating the probability of amplitude loss.
+
+    Raises:
+        TypeError: If the rate value is not a float.
+        ValueError: If the damping rate is not a correct probability.
+    """
+
+    def __init__(
+        self,
+        target: int,
+        rate: float,
+    ):
+        if rate > 1.0 or rate < 0.0:
+            raise ValueError("The damping rate is not a correct probability")
+        K0: Tensor = torch.tensor([[1, 0], [0, sqrt(1 - rate)]], dtype=DEFAULT_MATRIX_DTYPE)
+        K1: Tensor = torch.tensor([[0, sqrt(rate)], [0, 0]], dtype=DEFAULT_MATRIX_DTYPE)
+        Kraus_AmplitudeDamping: list[Tensor] = [K0, K1]
+        super().__init__(Kraus_AmplitudeDamping, target, rate)
+
+
+class PhaseDamping(Noise):
+    """
+    Initialize the PhaseDamping gate.
+
+    The phase damping channel is defined as:
+
+    .. math::
+        \\rho \\Rightarrow K_0 \\rho K_0^{\\dagger} + K_1 \\rho K_1^{\\dagger}
+
+     with:
+
+    .. code-block:: python
+
+        K0 = [[1, 0], [0, sqrt(1 - rate)]]
+        K1 = [[0, 0], [0, sqrt(rate)]]
+
+    Args:
+        target (int): The index of the qubit being affected by the noise.
+        rate (float): The damping rate, indicating the probability of phase damping.
+
+    Raises:
+        TypeError: If the rate value is not a float.
+        ValueError: If the damping rate is not a correct probability.
+    """
+
+    def __init__(
+        self,
+        target: int,
+        rate: float,
+    ):
+        if rate > 1.0 or rate < 0.0:
+            raise ValueError("The damping rate is not a correct probability")
+        K0: Tensor = torch.tensor([[1, 0], [0, sqrt(1 - rate)]], dtype=DEFAULT_MATRIX_DTYPE)
+        K1: Tensor = torch.tensor([[0, 0], [0, sqrt(rate)]], dtype=DEFAULT_MATRIX_DTYPE)
+        Kraus_GeneralizeAmplitudeDamping: list[Tensor] = [K0, K1]
+        super().__init__(Kraus_GeneralizeAmplitudeDamping, target, rate)
+
+
+class GeneralizeAmplitudeDamping(Noise):
```

Comment:
Not sure about this one. Feel free to ignore.

```suggestion
class GeneralizedAmplitudeDamping(Noise):
```

Reply:
Changed.